packages/venia-concept/src/RootComponents/Product/Product.js

```diff
@@ -51,13 +58,17 @@ class Product extends Component {
                     if (loading) return loadingIndicator;
 
                     const product = data.productDetail.items[0];
-
-                    return (
-                        <ProductFullDetail
-                            product={this.mapProduct(product)}
-                            addToCart={this.props.addItemToCart}
-                        />
-                    );
+                    const productData = this.mapProduct(product);
```

Comment:
Here is the result of the query for a product that is out of stock:

```json
{
  "data": {
    "productDetail": {
      "items": []
    }
  }
}
```

Therefore `const product = data.productDetail.items[0]` will result in `product` being `undefined` here and your conditional can just be `if (product) { ...`